src/Traits/Filterable.php

```diff
@@ -79,6 +81,7 @@ private function bootFilter(): void
             return (new FilterList())->only($this->getFilters());
         });
 
+        app()->when($this->defaultFilterResolverClass)->needs(Model::class)->give(fn () => $this);
```

Comment:
There should be one single resolver class during the lifecycle of any request. remove this line and update `OrFilter` to accept the replaced class.

Reply:
Seems like I'll be updating it during the day.
I wanted to keep it simple.
But agree with You.

Reply:
@abbasudo I'll make separate PR for it.